go/teams/chain.go

```diff
@@ -2248,6 +2277,48 @@ func (t *teamSigchainPlayer) parseTeamSettings(settings *SCTeamSettings, newStat
 	return nil
 }
 
+func (t *teamSigchainPlayer) parseTeamBotSettings(bots *[]SCTeamBot, newState *TeamSigChainState) error {
+
+	for _, bot := range *bots {
+		// Bots listed here must have the RESTRICTEDBOT role
+		role, err := newState.GetUserRole(bot.Bot.ToUserVersion())
+		if err != nil {
+			return err
+		}
+		if !role.IsRestrictedBot() {
+			return fmt.Errorf("found bot settings for %v. Expected role RESTRICTEDBOT, found %v", bot, role)
+		}
+
+		// Sanity check the triggers are valid
+		var convs, triggers []string
+		if bot.Triggers != nil {
+			for _, trigger := range *(bot.Triggers) {
+				if _, err := regexp.Compile(trigger); err != nil {
+					return err
+				}
+			}
+			triggers = *(bot.Triggers)
+		}
+		// Sanity check the conversation IDs are well formed
+		if bot.Convs != nil {
+			for _, convID := range *(bot.Convs) {
+				if _, err := chat1.MakeConvID(convID); err != nil {
+					return err
+				}
+			}
+			convs = *(bot.Convs)
+		}
+		newState.inner.Bots[bot.Bot.ToUserVersion()] = keybase1.TeamBotSettings{
+			Cmds:     bot.Cmds,
+			Mentions: bot.Mentions,
+			Joins:    bot.Joins,
+			Triggers: triggers,
+			Convs:    convs,
+		}
+	}
+	return nil
+}
```

Comment:
This checker looks correct. However, it might be easier to update this feature (bot settings) in the future if it wasn't picky here. And errored on reading bot settings rather than loading the team. Up to you